### PR TITLE
feat(config)!: * pattern matches everything

### PR DESCRIPTION
**Pull request by rarkins**
## Changes

Updates matchRegExOrGlob() so that '*' matches everything

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/string-pattern-matching.md [hunk 1]
```diff
@@ -76,6 +108,9 @@ For example, the pattern `["/^abc/", "!/^abcd/", "!/abce/"]`:
 - matches `"abc"` and `"abcf"`
 - does _not_ match `"foo"`, `"abcd"`, `"abce"`, or `"abcdef"`
 
+If you find yourself in a situation where you need to positive-match a string which starts with `!`, then you need to do so using a regular expression pattern.
+For example, `["/^!abc/$"]` will positively match against the string `"!abc"`.
+
```
Comment (rarkins) [inline docs/usage/string-pattern-matching.md:113]: This is not directly related but was not properly documented before so I prefer to combine it with this PR

File: docs/usage/string-pattern-matching.md [hunk 2]
```diff
@@ -5,6 +5,38 @@ Renovate string matching syntax for some configuration options allows you, as us
 - [`minimatch`](https://github.com/isaacs/minimatch) glob patterns, including exact strings matches
 - regular expression (regex) patterns
 
+The following fields support this pattern matching:
+
+- `allowedEnv`
```
Comment (secustor) [inline docs/usage/string-pattern-matching.md:10]: We should make this a field in options.ts when we have migrated. This would make the validation also more reliable and prevents us from maintaining this manually.

**Review by secustor — DISMISSED** [review]

File: docs/usage/string-pattern-matching.md [hunk 3]
```diff
@@ -76,6 +108,9 @@ For example, the pattern `["/^abc/", "!/^abcd/", "!/abce/"]`:
 - matches `"abc"` and `"abcf"`
 - does _not_ match `"foo"`, `"abcd"`, `"abce"`, or `"abcdef"`
 
+If you find yourself in a situation where you need to positive-match a string which starts with `!`, then you need to do so using a regular expression pattern.
+For example, `["/^!abc/$"]` will positively match against the string `"!abc"`.
```
Comment (viceice) [inline docs/usage/string-pattern-matching.md:112]:
```suggestion
For example, `["!/^abc$/"]` will positively match against the string `"!abc"`.
```
typo?

Comment (rarkins) [reply docs/usage/string-pattern-matching.md:112]: Not a typo

Comment (rarkins) [reply docs/usage/string-pattern-matching.md:112]: It's intended to show how to match a string which starts with ! so the pattern needs to be inside the / /

Comment (viceice) [reply docs/usage/string-pattern-matching.md:112]: but the dollar needs to be inside the regex

**Review by viceice — COMMENTED** [review]
updated comment

**Review by secustor — APPROVED** [review]
